tools/lintdiff.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+"""
+Drop in replacement for golangci-lint that runs it only on changed packages.
+
+Changes are calculated as diff against HEAD with a fallback to diff against main.
```

Comment:
Should the default be against main? When working on a PR, you'd want to lint your entire PR / branch rather than just uncommitted changes.

Reply:
Good point. Made script default to diff to main. HEAD is available via -h/HEAD option.